src/main/java/bhs/devilbotz/commands/DriveCommand.java

```diff
@@ -53,17 +54,32 @@ public void initialize() {}
    */
   @Override
   public void execute() {
+    // Clamp the values to the range [0.95, 0.95] to allow the PID loop some room to work
+    // TODO: replace clamp with dynamic range
+    double speed = MathUtil.clamp(this.speed.getAsDouble(), -0.95, 0.95);
+    double rot = MathUtil.clamp(this.rot.getAsDouble(), -0.95, 0.95);
```

Comment:
Can you explain how clamping the speed/rot gives the PID loop room to work?

Reply:
If going at full speed, the PID loop is not able to compensate the left and the right if on side is running slower, as both would already be at 100%

Reply:
Now that I understand these are the raw joystick values, it makes sense to me.